### ISSUE TITLE
Mod updates and auto updates

### ISSUE THREAD

**Issue by pinedours** [arkmanager/ark-server-tools#901]
Hello! 
I ve a problem when I m doing  checkmodupdate,
I ve always the answer  
```
Mod "idofthemod" doesn't exist in the steam workshop
```
So I suppose that my cron for automatically update mods doesn't work too?

**Comment by Malie73** [comment]
same here, since yesterday:

    ark@ark:~$ arkmanager checkmodupdate
    Running command 'checkmodupdate' for instance 'aberration'
    Mod 1091147617 doesn't exist in the steam workshop
    ...
    Mod 497432858 doesn't exist in the steam workshop

**Comment by pinedours** [comment]
Ok so it s not me..! Same when I start the server he does'nt check the mods anymore..

**Comment by wweiland** [comment]
Steam apparently changed the way that mods are downloaded.  Space Engineers is having the same problem.

See https://forum.keenswh.com/threads/dedicated-server-unable-to-download-mods.7399017/ for details.

**Comment by wweiland** [comment]
This is causing my servers to segfault while starting up.

**Comment by klightspeed** [comment]
v1.6.37 disables the mod update check.  Unfortunately this means that the output of `checkmodupdate` is less useful, and the program is forced to attempt to download all mods on every update cycle.

**Comment by wweiland** [comment]
I'm working now. Thank you again.  Hopefully Valve will see the error in their ways and fix the issue.

**Comment by Chingis-Khan** [comment]
Thanks @klightspeed for the quick fix :+1:

**Comment by wweiland** [comment]
Is there a possible long term fix in the works?

**Comment by klightspeed** [comment]
Without the ability to get the most recent mod version from Steam, it isn't possible to determine if a mod is up-to-date without trying to download it.

**Comment by klightspeed** [comment]
Looking at the output of `curl -s -d "itemcount=1&publishedfileids[0]=1093831470" http://api.steampowered.com/ISteamRemoteStorage/GetPublishedFileDetails/v1`, it looks like Valve removed the `hcontent_file` field, so the item's manifest cannot be used to check for updates.

This leaves us with the `time_updated` field to compare against the acf's `timeupdated` field.

**Comment by klightspeed** [comment]
v1.6.38 reinstates the mod update check, using the `time_updated` field.

**Comment by Malie73** [comment]
works flawless! Thanks

**Comment by Hulth** [comment]
I've had issues with arkmanager not detecting mod updates, it has happened twice with Classic flyers and once with Structures plus. (cronjob on `arkmanager update --update-mods --saveworld --ifempty @all`)

I'm running the latest arkmanager-version on master (1.6) and the code has the fixes from https://github.com/FezVrasta/ark-server-tools/commit/95ac5a61e115832f42c4bba88b09a1fec6886f1c

Only way to get the mods updated is using `arkmanager installmod 1234 --validate`

Unfortunately I didn't manually run the curl-command on the api so can't say what the values was at the time, but I'll do that next time and report any findings here. Thinking if it could be some caching somewhere? First issue was 21st December 2017

**Comment by Hulth** [comment]
Lol, as I wrote the previous comment 3 of my servers wasn't showing up because of a Classic Flyers update :P

Looked around, time_updated from the api was the same as timeupdated in appworkshop_346110.acf so naturally arkmanager didn't want to update it. Another check showed that all mod-files was recently modified in the staging-folder, but not in the actual server-folder. Did several refreshes in Winscp without the date changing. When I ran ls -la on the folder in a terminal it showed more recent dates on some files (and output was different in Winscp as well). Tried a restart of the server with no success.

Stopped the server, ran `arkmanger installmod 595711211 --validate @instance` and started the server. Then the mod was updated. Driving me nuts :(
